### DISCUSSION TITLE
How to get semicolon AST node information

### DISCUSSION THREAD

**Discussion by liuliangsir** [babel/babel#12332] [Q&A] [answered]
After using `@babel/standalone`, I find that the whole AST generated by the `@babel/standalone` above ignores(skips or omits) all the semicolon AST nodes. I would like to use information about the semicolon AST node. Is there any workaround to help me out?

**Comment by nicolo-ribaudo** [comment, marked as answer]
It's because semicolons aren't semantically meaningful, so they are not part of the AST. What you need is a [CST](https://github.com/cst/cst) (concrete syntax tree), which is not used/generated by Babel.

**Reply by watsom27** [reply]
While trying to write a custom plugin to be insert a profiling function call in a large number of functions i came across a situation where it seems like semicolons are semantically meaningful:

![image](https://user-images.githubusercontent.com/68954833/149188930-f6a853d5-602a-4ce9-89f2-b62ad9f53e00.png)

```
ERROR in ./someFile.ts 45:7
Module parse failed: Unexpected token (45:7)
File was processed with these loaders:
 * ./node_modules/babel-loader/lib/index.js
 * ./node_modules/esbuild-loader/dist/index.js
You may need an additional loader to handle the result of these loaders.
|     value: function adjustByMovingAnchor(newMovableAnchor2D, movableAnchor, wallsDependency = this.wallsDependencyBuilder.buildWallsDependencies(movableAnchor).confirmedWallsDependency) {
|       increment(adjustByMovingAnchor)
>       [...wallsDependency.keys()].forEach(wall => {
|         this.simplifyConnectedAuxiliaryPoints(wall, movableAnchor);
```

It seems in this case that I need to insert a semicolon after the function call. But i cant find where to do this in the docs and, judging by this thread, it doesn't look like its possible.

**Reply by nicolo-ribaudo** [reply]
You are probably inserting a `CallExpression` (`console.log("f")`) in a statement position. You need to wrap it with `t.expressionStatement(callExpressionNode)`.

**Reply by watsom27** [reply]
Thank you very much for your reply, this solved the issue
